addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoHandlerFactory.java

```diff
@@ -0,0 +1,171 @@
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.neeo.NeeoConstants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+
+/**
+ * The {@link NeeoHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+@Component(service = ThingHandlerFactory.class, immediate = true)
+public class NeeoHandlerFactory extends BaseThingHandlerFactory {
+
+    /** The {@link HttpService} used to register callbacks */
+    @NonNullByDefault({})
+    private HttpService httpService;
+
+    /** The {@link NetworkAddressService} used for ip lookup */
+    @NonNullByDefault({})
+    private NetworkAddressService networkAddressService;
+
+    /** The {@link DiscoveryServiceRegistry} used to initiate discovery */
+    @NonNullByDefault({})
+    private DiscoveryServiceRegistry discoveryServiceRegistry;
+
+    /** The scheduler used to schedule tasks */
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(NeeoConstants.THREADPOOL_ID);
+
+    /**
+     * Sets the {@link HttpService}.
+     *
+     * @param httpService the non-null {@link HttpService} to use
+     */
+    @Reference
+    protected void setHttpService(HttpService httpService) {
+        Objects.requireNonNull(httpService, "httpService cannot be null");
+        this.httpService = httpService;
+    }
+
+    /**
+     * Unsets the {@link HttpService}
+     *
+     * @param httpService the {@link HttpService} (not used in this implementation)
+     */
+    protected void unsetHttpService(HttpService httpService) {
+        this.httpService = null;
+    }
+
+    /**
+     * Sets the {@link NetworkAddressService}.
+     *
+     * @param networkAddressService the non-null {@link NetworkAddressService} to use
+     */
+    @Reference
+    protected void setNetworkAddressService(NetworkAddressService networkAddressService) {
+        Objects.requireNonNull(networkAddressService, "networkAddressService cannot be null");
+        this.networkAddressService = networkAddressService;
+    }
+
+    /**
+     * Unsets the {@link HttpService}
+     *
+     * @param networkAddressService the {@link HttpService} (not used in this implementation)
+     */
+    protected void unsetNetworkAddressService(NetworkAddressService networkAddressService) {
+        this.networkAddressService = null;
+    }
+
+    /**
+     * Sets the {@link NetworkAddressService}.
+     *
+     * @param discoveryServiceRegistry the non-null {@link NetworkAddressService} to use
+     */
+    @Reference
+    protected void setDiscoveryServiceRegistry(DiscoveryServiceRegistry discoveryServiceRegistry) {
```

Comment:
A binding should not interact with the `DiscoveryServiceRegistry`, please remove